activerecord/lib/active_record/relation.rb

```diff
@@ -920,10 +920,20 @@ def exec_queries(&block)
           end
 
           records = instantiate_records(rows, &block)
-          preload_associations(records) unless skip_preloading_value
 
-          records.each(&:readonly!) if readonly_value
-          records.each { |record| record.strict_loading!(strict_loading_value) } unless strict_loading_value.nil?
+          records.each do |record|
+            if null_relation? || !instance_variable_defined?("@association")
```

Comment:
Reaching out to the presence of an instance variable looks like a smell in my opinion. What is this trying to do?